src/main/java/com/vg/live/SuperLive.java

```diff
@@ -169,10 +175,17 @@ public static boolean isEOF(ByteBuffer buf) {
         int[] sampleDurations = fillArrayIfNull(trun.getSampleDurations(), sampleCount, trex.getDefaultSampleDuration());
         int[] samplesFlags = fillArrayIfNull(trun.getSamplesFlags(), sampleCount, trex.getDefaultSampleFlags());
 
+        TrackFragmentHeaderBox tfhd = new TrackFragmentHeaderBox(new Header(traf.getHeader().getFourcc()));
+        int sampleSize = checkedCast(tfhd.getDefaultSampleSize());
+        int sampleDuration = tfhd.getDefaultSampleDuration();
+        int sampleFlags = tfhd.getDefaultSampleFlags();
+
         for (int i = 0; i < sampleCount; i++) {
-            int sampleSize = checkedCast(trun.getSampleSize(i));
-            int sampleDuration = sampleDurations[i];
-            int sampleFlags = samplesFlags[i];
+            if (!trun.isFirstSampleFlagsAvailable()) {
```

Comment:
FIRST_SAMPLE_FLAG_PRESENT means there are no sample flags for all the samples AND special field exists for the flags of the first sample. It has nothing to do with size or duration. see https://github.com/chugai/mp4box.js/blob/e6af743f944f698b367b385dcb1f8363b780df4f/src/parsing/trun.js#L12-L17